### ISSUE TITLE
Feature request: Factor out or document a plane-building API

### ISSUE THREAD

**Issue by vebyast** [Tetragramm/Tetragramm.github.io#1]
I'm considering amusing myself by developing some tooling around building planes, things like a searchable design library or a "sane defaults" helper. How hard would it be to factor out something like a single function that accepts a design json and returns the design's summed and derived stats?

**Comment by Tetragramm** [comment]
Not too hard, since that's what I did.
Take a look at [impl/Aircraft.ts](https://github.com/Tetragramm/Tetragramm.github.io/blob/master/PlaneBuilder/impl/Aircraft.ts)  It contains the top-level Aircraft class.  
Among other things, the fromJSON, toJSON, GetStats, and GetDerivedStats functions.

**Comment by vebyast** [comment]
Fantastic, thanks!

Do you have plans to package this up for npm, or should I pull your repo in as a subpackage or something?

**Comment by Tetragramm** [comment]
Gotta be honest, Javascript/Typescript is not my favorite language, and I have no idea how to do that.  I would be happy to, if I could figure out how to keep the webpage working and have it export correctly.

**Comment by vebyast** [comment]
Ah, if you're not familiar with it, no problem; publishing to NPM wouldn't be the issue, it'd be all the satanic rituals involved in organizing JS so it can be imported. Not even kidding; I built [something almost identical to this](https://github.com/to-boldly-go/tbg-shipdesigner) for a different quest and that exact thing was where I got stuck. I'll go with a subpackage and see if it ends up being annoying enough to force me to figure out the JS ecosystem's approach to publishing stuff.
